syncthing-core/src/main/kotlin/net/syncthing/java/core/beans/DeviceAddress.kt

```diff
@@ -28,8 +27,11 @@ class DeviceAddress private constructor(val deviceId: String, private val instan
     val score = score ?: Integer.MAX_VALUE
     private val lastModified = lastModified ?: Date()
 
+    @Deprecated(message = "should use deviceIdObject instead")
     fun deviceId() = DeviceId(deviceId)
```

Comment:
Any reason you are leaving this in? It shouldn't be too hard to replace all usages with the new val.

Reply:
In this case, I'm not sure if I will replace the device id as string in the constructor with an object later which would require me to change it again everywhere, so I kept the old API but deprecated it.

For the other deprecated things: They mostly exist to keep old code working (until this other code is refactored too).